internal/controller/policyserver_controller_deployment.go

```diff
@@ -304,6 +345,88 @@ func (r *PolicyServerReconciler) adaptDeploymentSettingsForPolicyServer(policySe
 	}
 }
 
+func setOtelCertificateMounts(policyServerDeployment *appsv1.Deployment, otelCertificateSecret, otelClientCertificateSecret string) {
+	admissionContainer := &policyServerDeployment.Spec.Template.Spec.Containers[0]
+	defaultCertificateMountMode := int32(defaultOtelCertificateMountMode)
+
+	certificatePath := filepath.Dir(os.Getenv("OTEL_EXPORTER_OTLP_CERTIFICATE"))
+	if otelCertificateSecret != "" {
+		policyServerDeployment.Spec.Template.Spec.Volumes = append(policyServerDeployment.Spec.Template.Spec.Volumes, corev1.Volume{
```

Comment:
does this need to be idempotent? (same for `otelClientCertificateSecret`)

Reply:
This is idempotent, give we always start from an empty Deployment and add all the data defined inside of the PolicyServer `spec`, plus all the values we deem required.

@kubewarden/kubewarden-developers am I right?